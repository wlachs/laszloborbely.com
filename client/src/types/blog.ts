export type BlogPost = {
	urlHandle: string;
	title: string;
	author: string;
	summary: string;
	body: string;
	creationTime: Date;
};
