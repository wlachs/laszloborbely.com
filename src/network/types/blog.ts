export type BlogPostData = {
	author: string;
	body: string;
	creationTime: Date;
	id: string;
	summary: string;
	title: string;
};
