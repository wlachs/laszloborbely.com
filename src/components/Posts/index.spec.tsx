import { cleanup, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { afterEach, describe, expect, it } from 'vitest';

import { type BlogPostData } from '../../network/types/blog';
import { Posts } from './';

const post: BlogPostData = {
	author: 'me myself and I',
	body: 'body with a six-pack of beer',
	creationTime: new Date(0),
	summary: 'a summery summary',
	title: 'very very long test post',
	urlHandle: 'test-post',
};

describe('Posts component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const { container } = render(
			<MemoryRouter>
				<Posts data={[post]} isLoading={false} />,
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should display post metadata', () => {
		render(
			<MemoryRouter>
				<Posts data={[post]} isLoading={false} />,
			</MemoryRouter>,
		);
		expect(screen.getByText(post.title)).toBeDefined();
		expect(screen.getByText(post.summary)).toBeDefined();
	});

	it('should display message if there are no posts', () => {
		render(
			<MemoryRouter>
				<Posts data={[]} isLoading={false} />,
			</MemoryRouter>,
		);
		expect(
			screen.getByText('No posts found, come back later!'),
		).toBeDefined();
	});

	it('should display loading spinner if posts are not yet loaded', () => {
		render(
			<MemoryRouter>
				<Posts isLoading={true} />,
			</MemoryRouter>,
		);
		expect(screen.getByRole('status')).toBeDefined();
	});
});
