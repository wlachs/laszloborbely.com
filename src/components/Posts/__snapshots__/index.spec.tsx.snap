// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Posts component > should match snapshot 1`] = `
<div>
  <div
    class="posts w-100"
  >
    <a
      class="post w-100 py-3"
      href="/blog/test-post"
    >
      <div
        class="left-segment"
      >
        <h3
          class="title mb-1"
        >
          very very long test post
        </h3>
        <div
          class="summary mt-1"
        >
          a summery summary
        </div>
      </div>
      <div
        class="right-segment d-none d-md-block"
      >
        <div>
          54 years ago
        </div>
      </div>
    </a>
  </div>
  ,
</div>
`;
